apps/designer/app/designer/features/sidebar-left/panels/pages/settings.tsx

```diff
@@ -7,26 +7,119 @@ import {
   useId,
   styled,
   Flex,
-  toast,
+  toastNonFieldErrors,
   InputErrorsTooltip,
 } from "@webstudio-is/design-system";
-import { useFetcher } from "@remix-run/react";
-import { ChevronDoubleLeftIcon } from "@webstudio-is/icons";
+import { useFetcher, type Fetcher } from "@remix-run/react";
+import { ChevronDoubleLeftIcon, TrashIcon } from "@webstudio-is/icons";
+import { utils as projectUtils } from "@webstudio-is/project";
 import type { ZodError } from "zod";
 import { Header } from "../../lib/header";
-import { useEffect, useRef, useState } from "react";
+import { useRef, useState, useCallback } from "react";
 import { type Page } from "@webstudio-is/project";
+import { usePages } from "~/designer/shared/nano-states";
+import { useDebounce, useUnmount } from "react-use";
+import { useOnFetchEnd, usePersistentFetcher } from "~/shared/fetcher";
 
 const Group = styled(Flex, {
   marginBottom: "$3",
   gap: "$2",
   defaultVariants: { direction: "column" },
 });
 
-export type CreatePageData =
-  | { ok: true; page: Page }
+type FetcherData<Payload> =
+  | ({ ok: true } & Payload)
   | { errors: string | ZodError["formErrors"] };
 
+const normalizeErrors = (
+  errors: string | ZodError["formErrors"]
+): ZodError["formErrors"] =>
+  typeof errors === "string"
+    ? { formErrors: [errors], fieldErrors: {} }
+    : errors;
+
+type EditablePage = Omit<Page, "treeId" | "id">;
+
+const FormFields = ({
```

Comment:
would be probably easier to maintain if FormFields would be a controlled component, without the fetcher and then you would add the logic outside. This way you could for example render the form in storybook without fetching logic and also it wouldn't have much to debug